spec/lib/castle_spec.rb

```diff
@@ -0,0 +1,51 @@
+require 'rspec'
+require 'rspec/core'
+require 'json'
+require_relative '../../lib/castle.rb'
+require_relative '../../lib/room.rb'
+
+describe Castle do
+  before do
+    file = File.read('./game-layout.json')
+    @castle_data = JSON.parse(file)
```

Comment:
I'd construct the castle data in the test so the reader can see what's in it, rather than hiding it in a file. That will also get the file-reading and JSON noise out of here.